Test/JsonDiffTest.cs

```diff
@@ -0,0 +1,120 @@
+using c_sharp_json_diff;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Test
+{
+    public class JsonDiffTest
+    {
+        [SetUp]
+        public void Setup()
+        {
+        }
+
+        [Test, Category("Unit")]
+        [TestCaseSource("ProviderForNumericDiffTest")]
+        public void Given_TwoJsonStrings_When_NumericDiff_Then_ExpectedResultReturned(string left, string right, string expected)
+        {
+            JObject actualNumericDiff = JsonDiff.NumericDiff(left, right);
+            JObject expectedNumericDiff = JObject.Parse(expected);
+
+            Assert.AreEqual(expectedNumericDiff.Count, actualNumericDiff.Count);
```

Comment:
Instead of comparing the child count, use the `JObject.DeepEquals()` method. The count appears to only count direct children and does not recurse all the way down.